core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -74,22 +74,11 @@ class Project(
   }
   context.actorOf(Props(new ClassfileWatcher(searchService :: reTypecheck :: Nil)), "classFileWatcher")
 
-  // If the system property `ensime.server.async` is missing or not set to 
-  // `true` the old behaviour will be used.
-  private val handleAsyncConnections: Boolean = scala.util.Try(System.getProperty("ensime.server.async") == "true").toOption.getOrElse(false)
-
   def receive: Receive = {
-    if (handleAsyncConnections)
-      handleRequests
-    else
+    if (serverConfig.legacy.connectionInfoReq)
```

Comment:
I'm not too keen on this boolean check everywhere. Can you set the `receive` in the `preStart`?

Reply:
I'm not sure if `context.become` is clean in `preStart`. But I'll check that and refactor accordingly.

Reply:
this is akka... nothing is clean :smile: